agenda/presentation/src/main/java/com/gkp/agenda/presentation/compoments/DayItem.kt

```diff
@@ -0,0 +1,89 @@
+package com.gkp.agenda.presentation.compoments
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.drawBehind
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.gkp.core.designsystem.theme.TaskyOrange
+import com.gkp.core.designsystem.theme.TaskyTextFieldColor
+import com.gkp.core.designsystem.theme.TaskyTextHintColor
+import com.gkp.core.designsystem.theme.TaskyTheme
+
+@Composable
+fun DayItem(
+    modifier: Modifier = Modifier,
+    day: TaskyCalendarDay,
+    isDateSelected: Boolean = false,
+    onClick: () -> Unit = {}
+) {
+    Box(
+        modifier = modifier
+            .height(61.dp)
+            .width(40.dp)
+            .then(
+                if (isDateSelected) {
+                    Modifier.drawBehind {
+                        drawOval(
+                            color = TaskyOrange,
+                        )
+                    }
+                } else {
+                    Modifier
+                }
```

Comment:
It's quite ugly having to use conditional modifiers with `then` (still the correct way). I personally like to create a custom modifier like this:
```
fun Modifier.applyIf(
    condition: Boolean,
    modifier: Modifier.() -> Modifier
): Modifier {
    return this.then(
        if (condition) {
            modifier()
        } else Modifier
    )
}
```
You can then do
```
modifier = Modifier
    .applyIf(isDateSelected) {
        drawBehind { ... }
    }
```